lightning/src/ln/onion_utils.rs

```diff
@@ -1861,7 +1896,7 @@ impl HTLCFailReason {
 						failed_within_blinded_path: false,
 						hold_times: Vec::new(),
 						#[cfg(any(test, feature = "_test_utils"))]
-						onion_error_code: Some(*failure_code),
+						onion_error_code: Some(failure_reason.failure_code()),
```

Comment:
Especially for testing, the internal reason could be useful? Not necessarily in this PR.

Reply:
Good shout, I'll address in a follow up 👍